middleware/envcheckmiddleware.php

```diff
@@ -130,95 +136,41 @@ private function validateAndSetTokenBasedEnv() {
 		}
 	}
 
-	/**
-	 * Validates a token to make sure its linked to a valid resource
-	 *
-	 * Logic mostly duplicated from @see \OCA\Files_Sharing\Helper
-	 *
-	 * @fixme setIncognitoMode in 8.1 https://github.com/owncloud/core/pull/12912
-	 *
-	 * @param string $token
-	 *
-	 * @return array
-	 *
-	 * @throws CheckException
-	 */
-	private function getLinkItem($token) {
+    /**
+     * Validates a token to make sure its linked to a valid resource
+     *
+     * Uses Share 2.0
+     *
+     * @fixme setIncognitoMode in 8.1 https://github.com/owncloud/core/pull/12912
+     *
+     * @param string $token
+     * @return NotFoundResponse|Share\IShare
+     */
+    private function getLinkItem($token) {
 		// Allows a logged in user to access public links
 		\OC_User::setIncognitoMode(true);
 
-		$linkItem = Share::getShareByToken($token, false);
-
-		$this->checkLinkItemExists($linkItem);
-		$this->checkLinkItemIsValid($linkItem, $token);
-		$this->checkItemType($linkItem);
+        try {
+            $linkItem = $this->shareManager->getShareByToken($token);
+        } catch (ShareNotFound $e) {
+            return new NotFoundResponse();
```

Comment:
Not sure about that. I like the fact that we send different types of feedback depending on the error we get.
Did they throw it all away in files_sharing?